device/sycl/src/seeding/triplet_counting.sycl

```diff
@@ -56,56 +56,57 @@ class TripletCount {
                                            mb_idx, item);
 
         // get internal spacepoints for current bin
-        auto internal_sp_per_bin = internal_sp_device.bin(bin_idx);
-        auto& num_compat_spM_per_bin =
+        vecmem::device_vector internal_sp_per_bin = internal_sp_device.bin(bin_idx); 
+        const unsigned int& num_compat_spM_per_bin =
```

Comment:
This whole file should just be removed by this PR. Let's not make it nicer, just so that we could remove it right after...